Add search container style prop for GorgeousHeader

EricWiener opened the pull request:
Added a `searchBarStyle` prop to allow customization of the search input container.

EricWiener committed:
Updated README to reflect correct usage of GorgeousHeader

The header previously said to import `GorgeousHeader` as just `Gorgeous`, but this was causing an invalid import. Updated the README to reflect correct usage.

EricWiener committed:
GorgeousHeader: Added searchBarStyle prop to style search container

Added a search bar style prop that can be used to override the default
styles associated with the search bar container. This is useful if you
want to add shadow or decrease the margin between the text and search bar.

kuraydev commented on lib/src/components/GorgeousHeader/GorgeousHeader.js:
```diff
@@ -55,7 +56,7 @@ class GorgeousHeader extends Component {
           {subtitle}
         </Text>
-        <View style={styles.searchBarStyle}>
+        <View style={[styles.searchBarStyle, searchBarStyle]}>
```
You need to change it with this:

`searchBarStyle || styles.searchBarStyle`

Otherwise, it cannot overwrite the search bar's style :)

EricWiener replied:
I had left it this way on purpose so if you wanted to make small tweaks to the styles you could since the style values later on in the array override the earlier values. This makes it so you could do something like add shadow without overriding all the style values that exist. Do you still want me to change this?

kuraydev replied:
I think you should change it that way because with this way we let the developer to the whole customization and let the decision on them :)

EricWiener replied:
I just pushed again with the correction. Let me know if you need anything else.

kuraydev replied:
Lovely work! Thank you so much. I will release a new version and inform you.

EricWiener replied:
Glad to help. Sounds great

EricWiener committed:
Modified style overriding for search bar container

Changed the custom styling for the search bar container from being
additional styles applied ontop of the default styles to replacing
the default styles entirely.

WrathChaos commented:
@EricWiener I just released a new version :) 
Thank you!

## [Release: 0.5.1](https://github.com/WrathChaos/react-native-header-view/releases/tag/0.5.1)
